### PR TITLE
Fixed #30459 - Delegated hide/show JS toggle to parent div.

### PR DESCRIPTION
**Pull request by claudep**
https://code.djangoproject.com/ticket/30459

### CODE REVIEW

File: django/contrib/admin/static/admin/js/collapse.js [hunk 1]
```diff
@@ -30,23 +30,26 @@
...
-            var fieldset = closestElem(this, 'fieldset');
-            if (fieldset.classList.contains('collapsed')) {
-                // Show
-                this.textContent = gettext('Hide');
-                fieldset.classList.remove('collapsed');
-            } else {
-                // Hide
-                this.textContent = gettext('Show');
-                fieldset.classList.add('collapsed');
+            if (ev.target.nodeName === 'A' && ev.target.classList.contains('collapse-toggle')) {
```
Comment (carltongibson) [inline django/contrib/admin/static/admin/js/collapse.js]: Maybe [`Element.matches()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches) here?

```
if (ev.target.matches('.collapse-toggle`)) {
    ....
```

**Comment by carltongibson** [comment]
buildbot, test on selenium.

File: django/contrib/admin/static/admin/js/collapse.js [hunk 2]
```diff
@@ -30,23 +30,26 @@
...
+                } else {
+                    // Hide
+                    ev.target.textContent = gettext('Show');
+                    fieldset.classList.add('collapsed');
+                }
             }
         };
-        for (i = 0; i < toggles.length; i++) {
-            toggles[i].addEventListener('click', toggleFunc);
+        var inlineDivs = document.querySelectorAll('fieldset.module');
```
Comment (carltongibson) [inline django/contrib/admin/static/admin/js/collapse.js]: I widened this from `div.inline-group` to account for `admin_views.tests.SeleniumTests.test_collapsible_fieldset`

**Comment by carltongibson** [comment]
Hey @claudep. I added a selenium test case, and off the back of a related failure from that made a small adjustment to the where we attach the listener. Other than that I think it's ≈good to go. 

Can you cast your eye upon it? Thanks. 🙂

**Comment by carltongibson** [comment]
buildbot, test on selenium.

File: django/contrib/admin/static/admin/js/collapse.js (hunk 2 above)
Comment (claudep) [reply django/contrib/admin/static/admin/js/collapse.js]: Makes sense, many thanks for the test, which took probably more time to write than the fix itself!

Comment (carltongibson) [reply django/contrib/admin/static/admin/js/collapse.js]: Yeah. It did. 🙂

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**